barretenberg/cpp/src/barretenberg/vm2/simulation/concrete_dbs.cpp

```diff
@@ -98,6 +113,31 @@ void MerkleDB::nullifier_write(const FF& nullifier)
     (void)snapshot_after; // Silence unused variable warning when assert is stripped out
     // Sanity check.
     assert(snapshot_after == get_tree_roots().nullifierTree);
+
+    nullifier_counter++;
+}
+
+bool MerkleDB::note_hash_exists(const FF& note_hash) const
+{
+    auto [present, low_leaf_index] = raw_merkle_db.get_low_indexed_leaf(MerkleTreeId::NOTE_HASH_TREE, note_hash);
+    auto low_leaf_path = raw_merkle_db.get_sibling_path(MerkleTreeId::NOTE_HASH_TREE, low_leaf_index);
+    [[maybe_unused]] auto low_leaf_preimage = raw_merkle_db.get_leaf_preimage_nullifier_tree(low_leaf_index);
+
+    // TODO: Assert read
+    return true;
+}
+
+void MerkleDB::note_hash_write([[maybe_unused]] const FF& note_hash)
+{
+    // TODO: Do write
+    // AppendOnlyTreeSnapshot snapshot_after = note_hash_tree_check.write(
+    //     note_hash, low_leaf_hint.leaf, low_leaf_hint.index, low_leaf_hint.path, snapshot_before,
+    //     insertion_hint.path);
+    //
+    // (void)snapshot_after; // Silence unused variable warning when assert is stripped out
+    // // Sanity check.
+    // assert(snapshot_after == get_tree_roots().noteHashTree);
+    note_hash_counter++;
```

Comment:
IIUC these are not constrained here? If so please add a big comment/todo to constrain things. Do we even have gadgets for note hashes?

Reply:
yep, i've added a more obvious TODO - we dont have a note hash gadget yet.